Assets/HoloToolkit/Input/Scripts/Utilities/AttachToController.cs

```diff
@@ -42,90 +40,56 @@ public class AttachToController : MonoBehaviour
-            MotionControllerVisualizer.Instance.OnControllerModelUnloaded += DetachElementFromController;
+            base.OnEnable();
         }
 
-        protected virtual void OnDisable()
+        protected override void AddControllerTransform(MotionControllerInfo newController)
         {
-            if (MotionControllerVisualizer.IsInitialized)
-            {
-                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AttachElementToController;
-                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= DetachElementFromController;
-            }
-        }
-
-        protected virtual void OnDestroy()
-        {
-            if (MotionControllerVisualizer.IsInitialized)
+#if UNITY_WSA && UNITY_2017_2_OR_NEWER
+            if (!IsAttached)
             {
-                MotionControllerVisualizer.Instance.OnControllerModelLoaded -= AttachElementToController;
-                MotionControllerVisualizer.Instance.OnControllerModelUnloaded -= DetachElementFromController;
+                base.AddControllerTransform(newController);
             }
-        }
-
-        private void AttachElementToController(MotionControllerInfo newController)
-        {
-            if (!IsAttached && newController.Handedness == handedness)
-            {
-                if (!newController.TryGetElement(element, out elementTransform))
-                {
-                    Debug.LogError("Unable to find element of type " + element + " under controller " + newController.ControllerParent.name + "; not attaching.");
-                    return;
-                }
 
-                controller = newController;
+            SetChildrenActive(true);
 
-                SetChildrenActive(true);
+            // Parent ourselves under the element and set our offsets
+            transform.parent = ElementTransform;
+            transform.localPosition = positionOffset;
+            transform.localEulerAngles = rotationOffset;
 
-                // Parent ourselves under the element and set our offsets
-                transform.parent = elementTransform;
-                transform.localPosition = positionOffset;
-                transform.localEulerAngles = rotationOffset;
-                if (setScaleOnAttach)
-                {
-                    transform.localScale = scale;
-                }
+            if (setScaleOnAttach)
+            {
+                transform.localScale = scale;
+            }
 
-                // Announce that we're attached
-                OnAttachToController();
+            // Announce that we're attached
+            OnAttachToController();
 
-                IsAttached = true;
-            }
+            IsAttached = true;
+#endif
         }
 
-        private void DetachElementFromController(MotionControllerInfo oldController)
+        protected override void RemoveControllerTransform(MotionControllerInfo oldController)
```

Comment:
`base.RemoveControllerTransform` should be called in here, right?

Reply:
Yep. Added.